modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -2606,44 +2614,60 @@ TEST_P(Test_ONNX_layers, CumSum)
 }
 
 static void testYOLO(const std::string& weightPath, const std::vector<int>& refClassIds,
-                     const std::vector<float>& refScores, const std::vector<Rect2d>& refBoxes)
+                     const std::vector<float>& refScores, const std::vector<Rect2d>& refBoxes,
+                     Image2BlobParams imgParams, float conf_threshold = 0.3, float iou_threshold = 0.5,
+                     double scores_diff = 1e-5, double boxes_iou_diff = 1e-4, const std::string test_name = "")
 {
     std::string imgPath = _tf("../dog_orig_size.png");
 
-    Size targetSize{640, 640};
-    float conf_threshold = 0.3;
-    float iou_threshold = 0.5;
-
     Mat img = imread(imgPath);
-    Mat inp = blobFromImage(img, 1/255.0, targetSize, Scalar(0, 0, 0), true, false);
+
+    Mat inp = blobFromImageWithParams(img, imgParams);
 
     Net net = readNet(weightPath);
 
     net.setInput(inp);
     std::vector<Mat> outs;
     net.forward(outs, net.getUnconnectedOutLayersNames());
 
+    FilteredDetections result;
+    result = yoloPostProcessing(outs, conf_threshold, iou_threshold, test_name);
+
+    normAssertDetections(
+        refClassIds, refScores, refBoxes,
+        result.keep_classIds, result.keep_confidences, result.keep_boxes,
+        "", 0.0, scores_diff, boxes_iou_diff);
+}
+
+FilteredDetections yoloPostProcessing(std::vector<Mat>& outs, float conf_threshold, float iou_threshold, const std::string& test_name){
+
     // Retrieve
     std::vector<int> classIds;
     std::vector<float> confidences;
     std::vector<Rect2d> boxes;
+
+    if (test_name == "yolov8"){
+        cv::transposeND(outs[0], {0, 2, 1}, outs[0]);
+    }
+
     // each row is [cx, cy, w, h, conf_obj, conf_class1, ..., conf_class80]
-    for (auto preds : outs)
-    {
-        preds = preds.reshape(1, preds.size[1]);
+    for (auto preds : outs){
+
+        preds = preds.reshape(1, preds.size[1]); // [1, 8400, 85] -> [8400, 85]
+
         for (int i = 0; i < preds.rows; ++i)
         {
             // filter out non objects
-            float obj_conf = preds.row(i).at<float>(4);
+            float obj_conf = (test_name != "yolov8") ? preds.at<float>(i, 4) : 1.0f;
             if (obj_conf < conf_threshold)
                 continue;
 
-            // get class id and conf
-            Mat scores = preds.row(i).colRange(5, preds.cols);
+            Mat scores = preds.row(i).colRange((test_name != "yolov8") ? 5 : 4, preds.cols);
             double conf;
             Point maxLoc;
             minMaxLoc(scores, 0, &conf, 0, &maxLoc);
-            conf *= obj_conf;
+
+            conf = (test_name != "yolov8") ? conf * obj_conf : conf;
```

Comment:
There is a difference with a sample https://github.com/opencv/opencv/pull/24396. Does it mean that sample should be updated as well?

Reply:
From I quick glance, it seems as nothing needs to be changed since confidence score is taken as max over all scores. But the onnx the sample is using is difference from the one I am using

Reply:
Do you mean YOLOv8? Which one of the models is more convenient this way?

Reply:
For YOLOv8 there is no separate objectness score as for YOLOV5, 6,7, YOLOx. So I need to create it suing max class score. I did based on the [ultralytics](https://github.com/ultralytics/ultralytics) repository, which is the main repo for YOLOV8